contracts/SnappAuction.sol

```diff
@@ -179,20 +182,20 @@ contract SnappAuction is SnappBase {
         auctions[auctionIndex].size++;
     }
 
-    function placeMultiSellOrder(
-        uint8[] memory buyTokens,
-        uint8[] memory sellTokens,
-        uint128[] memory buyAmounts,
-        uint128[] memory sellAmounts
-    ) public {
-        uint numOrders = buyTokens.length;
-        require(numOrders == sellTokens.length, "numOrders != sellTokens length");
-        require(numOrders == buyAmounts.length, "numOrders != buyAmounts length");
-        require(numOrders == sellAmounts.length, "numOrders != sellAmounts length");
+    function placeMultiSellOrder(bytes memory packedOrders) public {
+        // Note that this could result failure of all orders if even one fails.
+        require(packedOrders.length % 27 == 0, "Each order should be packed in 27 bytes!");
+        for (uint i = 0; i < packedOrders.length / 27; i++) {
+            bytes memory orderData = packedOrders.slice(27*i, 27*(i+1));
+            unpackAndVerifyOrderData(orderData);
+            orderBatchUpdate();
 
-        for (uint i = 0; i < numOrders; i++) {
-            // onlyRegistered() enforced by this function call
-            placeSellOrder(buyTokens[i], sellTokens[i], buyAmounts[i], sellAmounts[i]);
+            bytes32 nextAuctionHash = sha256(abi.encodePacked(auctions[auctionIndex].shaHash, orderData));
+            auctions[auctionIndex].shaHash = nextAuctionHash;
+
+            emit PackedSellOrder(auctionIndex, auctions[auctionIndex].size, orderData);
```

Comment:
Can we for now just unpack the order and call the original `postSellOrder` function n times? Otherwise let's refactor postSellOrder to call into `placeMultiSellOrder`. At the end I'd really just like to maintain one unreserved sell order event in the listener.

Reply:
No, because I can't get `uint128` from the unpacking function.

Reply:
Yeah, it is easier to refactor `postSellOrder`, agreed.

Reply:
Cool, but we have a gas issue to deal with first.
I have concluded that either 
- commenting `unpackAndVerifyOrderData(orderData);` or
- replacing `bytes32 nextAuctionHash = sha256(abi.encodePacked(auctions[auctionIndex].shaHash, orderData));` with `bytes32 nextAuctionHash = 0x0`

would both work...

```js

    function placeMultiSellOrder(bytes memory packedOrders) public onlyRegistered() {
        // Note that this could result failure of all orders if even one fails.
        require(packedOrders.length % 28 == 0, "Each order should be packed in 28 bytes!");

        for (uint i = 0; i < packedOrders.length / 28; i++) {
            bytes memory orderData = packedOrders.slice(28*i, 28*(i+1));
            unpackAndVerifyOrderData(orderData);
            orderBatchUpdate();

            bytes32 nextAuctionHash = sha256(abi.encodePacked(auctions[auctionIndex].shaHash, orderData));
            auctions[auctionIndex].shaHash = nextAuctionHash;

            emit PackedSellOrder(auctionIndex, auctions[auctionIndex].size, orderData);
            // Only increment size after event (so it is emitted as an index)
            auctions[auctionIndex].size++;
        }
    }
```

Reply:
So is this related to the number of orders you are trying to place? As in does it work if you call `placeMultiSellOrder` with 0 bytes? At how many orders does it start failing? 

I guess we have to do some debugging to identify what is costing how much gas. You can use `it.only` and `truffle test -- gas` to effectively narrow down which call is causing how much gas.

Reply:
> So is this related to the number of orders you are trying to place? As in does it work if you call placeMultiSellOrder with 0 bytes? At how many orders does it start failing?

Unfortunately, the payload is not the problem here. This is a problem with contract size during migration and I haven't even started trying to unit test this with real orders yet.

We are already very well aware of the gas consumption of this contract on deployment. The following is the gas cost for deployment of the current version of the contract

```
   Deploying 'SnappAuction'
   ------------------------
   > transaction hash:    0x0026525ad588bd74b56e63ef6a43101f24f2b962bd1023221d2c9ad9a56d6f60
   > Blocks: 0            Seconds: 0
   > contract address:    0x9A2400BDeE35f01Deb6d6Cbdd24f0d2eAF9E92D5
   > account:             0xa0c455c7e343f2799E26f085bA8Ee671DDd2434A
   > balance:             99.69985876
   > gas used:            7010022
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.14020044 ETH
```